lucene/core/src/java/org/apache/lucene/index/ByteVectorValues.java

```
@@ -64,6 +65,18 @@ public static void checkField(LeafReader in, String field) {
     }
   }
 
+  /**
+   * Returns a new iterator that maps the provided docIds to the vector ordinals.
+   *
+   * <p>This method assumes that all docIds have corresponding ordinals.
+   *
+   * @lucene.internal
+   * @lucene.experimental
+   */
+  public DocIdSetIterator convertDocIdsToVectorOrdinals(DocIdSetIterator docIds) {
+    return docIds;
```

Wondering if this could move to the `KnnVectorValues` base class and be abstract and/or if a default implementation here (and in `FloatVectorValues`) should be `throw new UnsupportedOperationException();` i.e. a mapping needs to be provided by the implementations of this abstract class?

---

FWIW, I am against default implementations that throw, just require the interface and thus catch it at compile time.